elixir/apps/domain/lib/domain/events/hooks/accounts.ex

```diff
@@ -1,13 +1,42 @@
 defmodule Domain.Events.Hooks.Accounts do
+  alias Domain.PubSub
+  require Logger
+
   def on_insert(_data) do
     :ok
   end
 
-  def on_update(_old_data, _data) do
-    :ok
+  def on_update(old_data, data) do
+    with {:ok, account_id} <- Map.fetch(data, "id"),
+         {:ok, old_config} <- Map.fetch(old_data, "config"),
+         {:ok, config} <- Map.fetch(data, "config") do
+      if old_config != config do
+        broadcast(account_id, :config_changed)
+      else
+        :ok
+      end
+    else
+      :error ->
+        Logger.error("Expected keys not found in data",
+          old_data: inspect(old_data),
+          data: inspect(data)
+        )
+
+        :ok
+    end
   end
```

Comment:
I noticed there's a decent amount control flow in this function.  Would it be more of "the Elixir way" to utilize pattern matching here?  Something like:

```
def on_update(%{"config" => old_config} , %{"id" => account_id, "config" => config}) do
	if old_config != config do
		broadcast(account_id, :config_changed)
	else
		:ok
	end
end

def on_update(old_data, data) do
	Logger.error("Expected keys not found in data",
          old_data: inspect(old_data),
          data: inspect(data)
        )
	:ok
end
```

Reply:
Yeah, I like the idea of structs to allow for some level of type checking.

I can understand your concern about pattern matching in the function args if there are going to be a bunch of variations we're going to have to create.  If we can create structs, it seems like that might be able to help get some of the conditional checking out of these functions and let them deal with just the effects logic they need to deal with and worry about the verifying logic in another way.

Reply:
The right approach will make itself clear after I do a few more of these.